contracts/core/liquidators/twap-impl/TWAPAuction.sol

```diff
@@ -29,14 +40,389 @@ import { TwoAssetPriceBoundedLinearAuction } from "../impl/TwoAssetPriceBoundedL
+        TWAPState storage _twapAuction
+    )
+        internal
+        view
+    {
+        Auction.validateAuctionCompletion(_twapAuction.chunkAuction.auction);
+
+        require(
+            isRebalanceActive(_twapAuction),
+            "TWAPState.validateNextChunkAuction: TWAPAuction is finished."
+        );
+
+        require(
+            _twapAuction.lastChunkAuctionEnd.add(_twapAuction.chunkAuctionPeriod) <= block.timestamp,
+            "TWAPState.validateNextChunkAuction: Not enough time elapsed from last chunk auction end."
+        );
+    }
+
+    /**
+     * Checks if the amount of Sets still to be auctioned (in aggregate) is greater than the minimumBid
+     *
+     * @param _twapAuction                  TWAPAuction State object
+     */
+    function isRebalanceActive(
+        TWAPState storage _twapAuction
+    )
+        internal
+        view
+        returns (bool)
+    {
+        // Sum of remaining Sets in current chunk auction and order remaining
+        uint256 totalRemainingSets = _twapAuction.orderRemaining.add(
+            _twapAuction.chunkAuction.auction.remainingCurrentSets
+        );
+        
+        // Check that total remaining sets is greater than minimumBid
+        return totalRemainingSets > _twapAuction.chunkAuction.auction.minimumBid;
+    }
+
+    /**
+     * Calculates chunkSize of auction in current Set terms
+     *
+     * @param _totalSetAmount       Total amount of Sets being auctioned
+     * @param _rebalanceVolume      The total value being auctioned
+     * @param _usdChunkSize         Value of chunk size
+     */
+    function calculateChunkSize(
+        uint256 _totalSetAmount,
+        uint256 _rebalanceVolume,
+        uint256 _usdChunkSize
+    )
+        internal
+        view
+        returns (uint256)
+    {
+        // Since solidity rounds down anything equal to 1 will require at least one auction
+        // equal to the usdChunkSize
+        if (_rebalanceVolume.div(_usdChunkSize) >= 1) {
+            return _totalSetAmount.mul(_usdChunkSize).div(_rebalanceVolume);
+        } else {
+            return _totalSetAmount;
+        }
+    }
+
+    /**
+     * Get asset pair hash from passed collateral.
+     *
+     * @param _currentSet                   The Set to rebalance from
+     * @param _nextSet                      The Set to rebalance to
+     */
+    function getAssetPairHashFromCollateral(
+        ISetToken _currentSet,
+        ISetToken _nextSet
+    )
+        internal
+        view
+        returns (bytes32)
+    {
+        // Get set components
+        address[] memory currentSetComponents = _currentSet.getComponents();
+        address[] memory nextSetComponents = _nextSet.getComponents();
+
+        address[] memory assetPairComponents = currentSetComponents.union(nextSetComponents);
+
+        return getAssetPairHash(assetPairComponents[0], assetPairComponents[1]);
+    }
+
+    /**
+     * Parse passed Liquidator data
```

Comment:
If the comment doesn't say more than the function name itself, don't really need it